tests/test_spike_locations.py

```diff
@@ -0,0 +1,71 @@
+# 8/23/22
+# https://figurl.org/f?v=gs://figurl/spikesortingview-8&d=sha1://d8ed07c1d4653f0e22c80471f4c921c4aa7d30fc&label=test_spike_locations
+
+import numpy as np
+from typing import List
+import sortingview as sv
+import sortingview.views as vv
+import spikeinterface as si
+import spikeinterface.extractors as se
+
+
+def main():
+    recording, sorting = se.toy_example(num_units=12, num_channels=10, duration=300, seed=0)
+
+    R = sv.copy_recording_extractor(recording, serialize_dtype='float32')
+    S = sv.copy_sorting_extractor(sorting)
+
+    view = test_spike_locations(recording=R, sorting=S, hide_unit_selector=False)
+
+    url = view.url(label='test_spike_locations')
+    print(url)
+
+def test_spike_locations(*, recording: si.BaseRecording, sorting: si.BaseSorting, hide_unit_selector: bool=False):
+    channel_locations = recording.get_channel_locations().astype(np.float32)
+    xmin = np.min(channel_locations[:, 0])
+    xmax = np.max(channel_locations[:, 0])
+    ymin = np.min(channel_locations[:, 1])
+    ymax = np.max(channel_locations[:, 1])
+    xspan = xmax - xmin
+    yspan = ymax - ymin
+    if xmax <= xmin:
+        xmin = xmin - 12
+        xmax = xmax + 12
+    if ymax <= ymin:
+        ymin = ymin - 12
+        ymax = ymax + 12
+    # expand ranges
+    xspan = xmax - xmin
+    yspan = ymax - ymin
+    xmin = xmin - xspan * 0.2
+    xmax = xmax + xspan * 0.2
+    ymin = ymin - yspan * 0.2
+    ymax = ymax + yspan * 0.2
+
+    rng = np.random.default_rng(2022)
+    items: List[vv.SpikeLocationsItem] = []
+    for unit_id in sorting.get_unit_ids():
+        spike_times_sec = np.array(sorting.get_unit_spike_train(unit_id=unit_id)) / sorting.get_sampling_frequency()
+        center_x = rng.uniform(xmin, xmax) # fake center for unit
+        center_y = rng.uniform(ymin, ymax) # fake center for unit
+        items.append(
+            vv.SpikeLocationsItem(
+                unit_id=unit_id,
+                spike_times_sec=spike_times_sec.astype(np.float32),
+                x_locations=rng.normal(center_x, 6, spike_times_sec.shape).astype(np.float32), # fake locations
+                y_locations=rng.normal(center_y, 6, spike_times_sec.shape).astype(np.float32)  # fake locations
```

Comment:
Is there any particular significance to setting the standard deviation to 6? I'm guessing this is related to spreading the x and y ranges by 24 earlier?

Reply:
This is all just a test... was just trying to make the figure look good.